apps/curation_front_end/src/components/CurationForm.tsx

```diff
@@ -3,9 +3,193 @@
 // SPDX-License-Identifier: MIT
 
 import React from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { userAPI } from "validation-helpers";
+import {
+    Form,
+    FormControl,
+    FormField,
+    FormItem,
+    FormMessage,
+} from "@/components/ui/form";
+import { Button } from "@/components/ui/button";
+import { ZodiosBodyByPath, ZodiosResponseByPath } from "@zodios/core";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import CurationSPDX from "./CurationSPDX";
+import CurationDB from "./CurationDB";
+import CurationLicense from "./CurationLicense";
+import { userHooks } from "@/hooks/zodiosHooks";
 
-const CurationForm = () => {
-    return <div>CurationForm</div>;
+const curationFormSchema = z.object({
+    fileSha256: z.string(),
+    detectedLicenseExpressionSPDX: z.string().optional(),
+    concludedLicenseExpressionSPDX: z.string(),
+    concludedLicenseSPDX: z.string().optional(),
+    concludedLicenseDB: z.string().optional(),
+    concludedLicenseList: z.string().optional(),
+    comment: z.string().optional(),
+    contextPurl: z.string(),
+});
+
+type CurationFormType = z.infer<typeof curationFormSchema>;
+
+type DataType = ZodiosResponseByPath<typeof userAPI, "post", "/file">;
+type LicenseConclusionPostData = ZodiosBodyByPath<
+    typeof userAPI,
+    "post",
+    "/license-conclusion"
+>;
+
+type Props = {
+    purl: string | undefined;
+    fileData: DataType;
+};
+
+const CurationForm = ({ purl, fileData }: Props) => {
+    const defaultValues: Partial<CurationFormType> = {
+        concludedLicenseExpressionSPDX: "",
+        detectedLicenseExpressionSPDX:
+            fileData?.licenseConclusions[0]?.detectedLicenseExpressionSPDX ??
```

Comment:
Detected expression should come from license finding (fileData?.licenseFindings[0]?.licenseExpressionSPDX)

Reply:
Yes, fixing.